src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -61,6 +56,14 @@ public void Initialize(EntityInfoInternal entity, Func<IncomingMessageDetailsInt
         }
 
         public void Start()
+        {
+            if (Interlocked.Increment(ref refCount) == 1)
```

Comment:
Get your point. The problem is that if two concurrent callers try to start the first will start it and if it takes long the second one will "think" it is already started. I don't think this is a real problem here because the receive notifier is a push-based approach. Thus a client of the notifier cannot do anything meaningful with it other than just wait that at some point in time something will be pushed to the delegates that were passed in. Making any assumptions about when that will happen is a violation of the nature of this component (plus in real life scenarios nothing could be in the queue thus making assumptions that something gets pushed as soon as you called start seems wrong)